### PR TITLE
feat(ci): add codecov

### PR DESCRIPTION
**Pull request by Joshua-Anderson**

### CODE REVIEW

**Comment by Joshua-Anderson** [comment]
@vdice Any idea why Jenkins is having syntax errors on this pull request?

**Comment by vdice** [comment]
@Joshua-Anderson hmmm I don't know why it would complain... I'm not aware of the `bash <(<bash script>)` syntax and/or how it differs/compares to the classic `<bash script> | bash` approach... but perhaps try the latter approach?

File: Makefile [hunk 1]
```diff
@@ -91,17 +91,18 @@ setup-gotools:
...
 	@for i in $(addsuffix /...,$(GO_PACKAGES)); do \
 		${DEV_ENV_CMD} golint $$i; \
 	done
 
-test-unit:
+test: test-style
 	${DEV_ENV_PREFIX_CGO_ENABLED} ${DEV_ENV_IMAGE} sh -c '${GOTEST} $$(glide nv)'
 
+test-cover: test-style
+	${DEV_ENV_PREFIX_CGO_ENABLED} ${DEV_ENV_IMAGE} test-cover.sh
```
Comment (bacongobbler) [inline Makefile]: Where is this script located?

Comment (Joshua-Anderson) [inline Makefile]: It's inside the [docker-go-dev container](https://github.com/deis/docker-go-dev/blob/master/rootfs/usr/local/bin/test-cover.sh)

**Comment by Joshua-Anderson** [comment]
@vdice The script works for me locally and `test-cover` does successfully generate a coverage support (`coverage.txt`). However, in CI, codecov says it can't find any coverage files. Any ideas? Could this be related to the file being created with root permissions? If so, why doesn't travis-ci have this problem?

File: Makefile (hunk 1 above)
Comment (bacongobbler) [inline Makefile]: TIL. Cool!

**Comment by Joshua-Anderson** [comment]
As you can see, the PR works! Ready for reviews!

**Comment by Joshua-Anderson** [comment]
This [passed e2e](https://ci.deis.io/job/workflow-test-pr/4300/console) but the result wasn't properly logged

**Pull request merged**